File: 75_Sort_Colors.py
class Solution(object):
    def sortColors(self, x):
        a, b, c = 0, 0, len(x) - 1
        while b <= c:
            if x[b] == 0: x[a], x[b], a, b = x[b], x[a], a+1, b+1
            elif x[b] == 2: x[b], x[c], c = x[c], x[b], c-1
            else: b += 1

File: 75_Sort_Colors.java
class Solution {
    public void sortColors(int[] y) {
        int[] f = new int[3];
        for (int z : y) f[z]++;
        int i = 0;
        for (int j = 0; j < 3; j++)
            while (f[j]-- > 0) y[i++] = j;
    }
}
